### PR TITLE
Improve account email receiving e2e test to fix flakiness

### LINKED ISSUE #47506: [Flaky] shopper/account-email-receiving.spec.js
should receive an email when creating an account 

Failed on: https://github.com/woocommerce/woocommerce/actions/runs/9095109166

### PR DESCRIPTION
**Pull request by veljkho**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #47506

Fixes flakiness to zero.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

```
pnpm test:e2e-pw tests/e2e-pw/tests/shopper/account-email-receiving.spec.js
```

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/tests/e2e-pw/tests/shopper/account-email-receiving.spec.js [hunk 1]
```diff
@@ -1,9 +1,25 @@
...
 const emailContent = '#wp-mail-logging-modal-content-body-content';
-const emailContentJson = '#wp-mail-logging-modal-format-json';
+const emailContentHtml = '#wp-mail-logging-modal-format-html';
+
+const now = Date.now();
+const username = `${ now }`;
+const email = `${ now }@example.com`;
+const firstName = `Customer`;
+const lastName = `the ${ now }th`;
+const role = 'Customer';
```
Comment (adimoldovan) [inline plugins/woocommerce/tests/e2e-pw/tests/shopper/account-email-receiving.spec.js]: By adding the user fixture, these should be moved in the fixture, and used in the test via the user object.


```
const user = {
	username: `${now}`
	email: `${ now }@example.com`
	...
}

```
and in the test:

```
...
await page.getByLabel( 'Username' ).fill( user.username );
...
```

**Review by adimoldovan — CHANGES_REQUESTED** [review]

File: plugins/woocommerce/tests/e2e-pw/tests/shopper/account-email-receiving.spec.js (hunk 1 above)
Comment (veljkho) [reply plugins/woocommerce/tests/e2e-pw/tests/shopper/account-email-receiving.spec.js]: That's a good catch! Done.

File: plugins/woocommerce/tests/e2e-pw/tests/shopper/account-email-receiving.spec.js [hunk 2]
```diff
@@ -1,20 +1,36 @@
...
+		console.log( `Deleting user ${ user.id }` );
+		await api.delete( `customers/${ user.id }`, { force: true } );
+	},
+} );
 
 test.describe(
 	'Shopper Account Email Receiving',
 	{ tag: [ '@payments', '@services' ] },
 	() => {
 		test.use( { storageState: process.env.ADMINSTATE } );
```
Comment (adimoldovan) [inline plugins/woocommerce/tests/e2e-pw/tests/shopper/account-email-receiving.spec.js]: This `test.use` is redundant, it's already set above. The same with the one in the next describe block.

**Review by adimoldovan — COMMENTED** [review]
Thanks for updating this @veljkho. I mentioned the redundant calls of `test.use` below, but other than that this looks good to me.

File: plugins/woocommerce/tests/e2e-pw/tests/shopper/account-email-receiving.spec.js (hunk 2 above)
Comment (veljkho) [reply plugins/woocommerce/tests/e2e-pw/tests/shopper/account-email-receiving.spec.js]: Ah, that's some leftover from the past. Removed.

**Review by adimoldovan — APPROVED** [review]

**Pull request merged**